python/lsst/ts/externalscripts/base_make_calibrations.py

```diff
@@ -444,32 +494,46 @@ async def call_pipetask(self, image_type, exposure_ids_dict):
             pipe_yaml = "findDefects.yaml"
             config_string = (
                 f"-j {self.config.n_processes} -i {self.config.input_collections_defects} "
-                f"-i {self.config.calib_collection} "
                 "--register-dataset-types "
                 f"{self.config.config_options_defects}"
             )
             exposure_ids = exposure_ids_dict["DARK"] + exposure_ids_dict["FLAT"]
-        elif image_type == "PTC":
-            pipe_yaml = "cpPtc.yaml"
+        elif image_type == "PTC" or image_type == "GAIN":
+            if image_type == "PTC":
+                pipe_yaml = "cpPtc.yaml"
+            else:
+                pipe_yaml = "cpPtc.yaml#gainFromFlatPairs"
             config_string = (
                 f"-j {self.config.n_processes} -i {self.config.input_collections_ptc} "
-                f"-i {self.config.calib_collection} "
                 "--register-dataset-types "
                 f"{self.config.config_options_ptc}"
             )
             exposure_ids = exposure_ids_dict["FLAT"]
         else:
             raise RuntimeError(
                 "Invalid image or calib type {image_type} in 'call_pipetask' function. "
-                "Valid options: ['BIAS', 'DARK', 'FLAT', 'DEFECTS', 'PTC']"
+                "Valid options: ['BIAS', 'DARK', 'FLAT', 'DEFECTS', 'PTC', 'GAIN']"
             )
 
+        # If we are using internal calibrations, place the
+        # calibration collection where they were certified
+        # before the input collections of each task
+        # (except for BIAS, which is the first step).
+        if self.config.generate_calibrations and image_type != "BIAS":
+            split_string = config_string.split()
+            # The third entry is where the input collections are.
+            split_string[3] = f"{self.config.calib_collection}," + split_string[3]
+            config_string = " ".join(split_string)
+
```

Comment:
I think I mentioned this before but I don't thing this if/else should be here. Again, if you add support to another camera, you would have to change this code here, which defeats the purpose of having a base class etc.. 

Instead make `pipeline_instrument` an abstract class property.